src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -72,25 +72,27 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
      * i.e., file:///path/to/bam.bam.
      * @param readFileName file to load
      * @param referencePath Reference path or null if not available. Reference is required for CRAM files.
+     * @param auth authentication information if using GCS.
      * @param intervals intervals of reads to include.
      * @return RDD of (SAMRecord-backed) GATKReads from the file.
      */
-    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final List<SimpleInterval> intervals) {
-        return getParallelReads(readFileName, referencePath, intervals, 0);
+    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final AuthHolder auth, final List<SimpleInterval> intervals) {
+        return getParallelReads(readFileName, referencePath, auth, intervals, 0);
     }
 
     /**
      * Loads Reads using Hadoop-BAM. For local files, bam must have the fully-qualified path,
      * i.e., file:///path/to/bam.bam.
      * @param readFileName file to load
      * @param referencePath Reference path or null if not available. Reference is required for CRAM files.
+     * @param auth authentication information if using GCS.
      * @param intervals intervals of reads to include. If <code>null</code> then all the reads (both mapped and unmapped) will be returned.
      * @param splitSize maximum bytes of bam file to read into a single partition, increasing this will result in fewer partitions. A value of zero means
      *                  use the default split size (determined by the Hadoop input format, typically the size of one HDFS block).
      * @return RDD of (SAMRecord-backed) GATKReads from the file.
      */
-    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final List<SimpleInterval> intervals, final long splitSize) {
-        SAMFileHeader header = getHeader(readFileName, referencePath, null);
+    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final AuthHolder auth, final List<SimpleInterval> intervals, final long splitSize) {
+        SAMFileHeader header = getHeader(readFileName, referencePath, auth);
```

Comment:
Can we just convert `getHeader()` to use an NIO Path in the GCS case, instead of passing around this `AuthHolder` object and calling into `BAMIO.openBAM()` from the dataflow library? If yes, I'd prefer to do that rather than spread use of the obsolete `AuthHolder` class further. 

@jean-philippe-martin do you concur?